finetuner-workflow/finetune-workflow.yaml

```diff
@@ -0,0 +1,300 @@
+      resources:
+        requests:
+          memory: 256Mi
+          cpu: "4"
+      volumeMounts:
+        - mountPath: "/{{workflow.parameters.pvc}}"
+          name: "{{workflow.parameters.pvc}}"
+    volumes:
+      - name: "{{workflow.parameters.pvc}}"
+        persistentVolumeClaim:
+           claimName: "{{workflow.parameters.pvc}}"
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: topology.kubernetes.io/region
+              operator: In
+              values:
+              - "{{workflow.parameters.region}}"
+
+
+  - name: model-finetuner
+    inputs:
+      parameters:
+        - name: run_name
+        - name: model
+        - name: dataset
+        - name: learn_rate
+        - name: epochs
+        - name: train_ratio
+        - name: eot
+        - name: pad
+        - name: bs
+        - name: bs_divisor
+        - name: seed
+        - name: output_path
+        - name: cache
+        - name: logs
+        - name: torch_cache
+        - name: checkpoint
+        - name: wandb_key
+    container:
+      image: docker.io/gooseai/finetuner:rc38
+      command: [ "/usr/bin/python3", "/app/finetuner.py" ]
+      args: ["--run_name", "{{inputs.parameters.run_name}}",
+             "--model", "{{inputs.parameters.model}}",
+             "--dataset", "{{inputs.parameters.dataset}}",
+             "--eot", "{{inputs.parameters.eot}}",
+             "--pad", "{{inputs.parameters.pad}}",
+             "--bs", "{{inputs.parameters.bs}}",
+             "--seed", "{{inputs.parameters.seed}}",
+             "--lr", "{{inputs.parameters.learn_rate}}",
+             "--tr", "{{inputs.parameters.train_ratio}}",
+             "--output", "{{inputs.parameters.output_path}}",
+             "--cache", "{{inputs.parameters.cache}}",
+             "--logs", "{{inputs.parameters.logs}}",
+             "--bs_divisor", "{{inputs.parameters.bs_divisor}}",
+             "--{{inputs.parameters.checkpoint}}"]
+      tty: true
+      env:
+      - name: WANDB_API_KEY
+        value: "{{inputs.parameters.wandb_key}}"
+      - name: PYTHONUNBUFFERED
+        value: "1"
+      - name: TORCH_EXTENSIONS_DIR
+        value: "{{inputs.parameters.torch_cache}}"
+      resources:
+        requests:
+          memory: 128Gi
+          cpu: "8"
+        limits:
+          nvidia.com/gpu: 1
+      volumeMounts:
+        - mountPath: "/{{workflow.parameters.pvc}}"
+          name: "{{workflow.parameters.pvc}}"
+    volumes:
+      - name: "{{workflow.parameters.pvc}}"
+        persistentVolumeClaim:
+           claimName: "{{workflow.parameters.pvc}}"
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: gpu.nvidia.com/vram
+              operator: In
+              values:
+              - "48"
```

Comment:
We should select one or multiple specific GPUs, it's less confusing to users imo.

Reply:
Done!

Reply:
Also parameterized the GPU, because, hey, why not?